x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -725,9 +732,23 @@ export class ObservabilityAIAssistantClient {
     return this.dependencies.knowledgeBaseService.getStatus();
   };
 
-  setupKnowledgeBase = (modelId: string | undefined) => {
-    const { esClient } = this.dependencies;
-    return this.dependencies.knowledgeBaseService.setup(esClient, modelId);
+  setupKnowledgeBase = async (modelId: string | undefined) => {
+    const { esClient, core, logger, knowledgeBaseService } = this.dependencies;
+
+    // setup the knowledge base
+    const res = await knowledgeBaseService.setup(esClient, modelId);
+
+    core
+      .getStartServices()
+      .then(([_, pluginsStart]) => {
+        logger.debug('Schedule semantic text migration task');
+        return scheduleSemanticTextMigration(pluginsStart);
```

Comment:
We can’t run the migration until the user creates the inference endpoint, which only happens when they click "Setup." We schedule the migrations here as well because the migration task set up at startup will time out after a couple of hours. If the user doesn’t click "Setup" within that time, the migrations won’t run.

We still need to run migrations at startup to handle cases where the user creates the inference endpoint but stops Kibana before the migrations have a chance to complete.